src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -38,10 +67,191 @@ private void SetContentDispositionHeader(ActionContext context, FileResult resul
             }
         }
 
-        private void SetContentType(ActionContext context, FileResult result)
+        private void SetAcceptRangeHeader(ActionContext context)
         {
             var response = context.HttpContext.Response;
-            response.ContentType = result.ContentType;
+            response.Headers[HeaderNames.AcceptRanges] = AcceptRangeHeaderValue;
+        }
+
+        // Internal for testing
+        internal static bool ComputeConditionalRequestHeaders(
```

Comment:
@Tratcher Could you review this one in particular? I tried to follow [this](https://tools.ietf.org/html/rfc7232#section-6)